TypeInjections/Tool/Tool.fs

```diff
@@ -149,21 +134,40 @@ module Tool =
 
         // if verb found, result = Parsed<obj> = map of options and values
         // if no verb found or help asked for, result = NotParsed<obj> = list of error messages
-        match result with
-        | :? (Parsed<obj>) as command ->
-            match command.Value with
-            | :? LocalOptions as opts -> runLocal opts
-            | :? GitOptions as opts -> runGit opts
-            | _ -> ()
-        // nothing to do if result=NotParsed<obj> because helpful error message
-        // was already thrown by ParseArgument
-        | :? (NotParsed<obj>) as errors ->
-            // see what the error is
-            match (getError errors.Errors) with
-            | :? BadVerbSelectedError -> failwith $"bad verb"
-            | :? HelpRequestedError -> failwith "user just wanted help!"
-            | _ -> failwith "unknown error, need to investigate"
-        | _ -> ()
 
+        // have runGit or runLocal change value of these variables
+        let input =
+            match result with
+            | :? (Parsed<obj>) as command ->
+                match command.Value with
+                | :? LocalOptions as opts -> runLocal opts
+                | :? GitOptions as opts -> runGit opts
+                | _ -> parsedOption ("", "", "", [])
+            // nothing to do if result=NotParsed<obj> because helpful error message
+            // was already thrown by ParseArgument
+            | :? (NotParsed<obj>) as errors ->
+                // see what the error is
+                match (getError errors.Errors) with
+                | :? BadVerbSelectedError -> failwith $"bad verb"
+                | :? HelpRequestedError -> failwith "user just wanted help!"
+                | _ -> failwith "unknown error, need to investigate"
+            | _ -> parsedOption ("", "", "", [])
+
+
+
+        if input.oldPath.Length > 0 then
+            let reporter = Utils.initDafny
+
+            (* TypecartProject's 2nd argument can be:
+            1. the absolute path to .typecartignore file
+            2. a list of regex values *)
```

Comment:
Seems like `ignore` can only be a list of strings (regex values) in our case right now. How do we support `typecartignore`?

Reply:
To support `typecartignore` we would need to input the path to an external typecart ignore file. I am not sure the usefulness of this, especially since Ruijie added the feature where `.typecartignore` files were automatically ran in they were within the project. I might want to drop support for the `.typecartignore` file